frontend/src/pages/Author.jsx

```diff
@@ -54,14 +94,20 @@ class Author extends Component {
             .then((httpResponse) => {
                 if (httpResponse.status === 200) {
                     httpResponse.json().then((results) => {
-                        const authorID = this.getloggedinAuthorIDandHost();
-                        let isFriends = false;
-                        for (let i = 0; i < results.friends.length; i++) {
-                            if (results.friends[i].id === authorID[0]){
-                                isFriends = true;
-                                break;
+                        let canFollow = true;
+                        if (results.friends) {
+                            for (let i = 0; i < results.friends.length; i++) {
+                                if (results.friends[i].id === authenticatedAuthorId){
+                                    canFollow = false;
+                                    break;
+                                }
                             }
+                        } else {
+                            // results.friends not available means it is a foreign friend
+                            // then we need to fetch our own profile to check if i am friend/following him already
```

Comment:
```suggestion
                            // then we need to fetch our own profile to check if I am friend/following him already
```
capitalization heeheexd